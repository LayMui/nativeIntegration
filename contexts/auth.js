import React, { useCallback, useEffect, useMemo, useState } from "react";
import { NativeUserManagementContext, useUserManagement } from "@yaradigitallabs/sh-user-management-native";
import { LOGIN_TYPES } from "../constants/routes";

const contextFactory = ({
  accessToken,
  login,
  logout,
  forceLogin,
  loginViaBrowser,
  logoutViaBrowser,
  forceLoginViaBrowser,
  setUserProfile,
  userProfile,
  updateUserProfile,
  isAuth0Used,
  saveCredentialsViaEmbedded,
  // startLoginViaEmbedded,
  completeLoginViaEmbedded,
  getToken,
  setEnabledAnalytics,
  selectedLoginType,
  changeSelectedLoginType,
} = {}) => ({
  accessToken,
  login,
  logout,
  forceLogin,
  loginViaBrowser,
  logoutViaBrowser,
  setUserProfile,
  userProfile,
  updateUserProfile,
  isAuth0Used,
  forceLoginViaBrowser,
  // saveCredentialsViaEmbedded,
  // startLoginViaEmbedded,
  // completeLoginViaEmbedded,
  getToken,
  setEnabledAnalytics,
  selectedLoginType,
  changeSelectedLoginType,
});

const context = React.createContext(contextFactory());

const AuthContext = ({ children }) => {
  const {
    login,
    logout,
    forceLogin,
    loginViaWebView,
    logoutViaWebView,
    forceLoginViaWebView,
    getUserProfile,
    updateUserProfile,
    saveCredentialsEmbedded,
    startLoginEmbedded,
    completeLoginEmbedded,
    getToken,
    setEnabledAnalytics,
    logoutViaEmbedded,
  } = useUserManagement();
  const [accessToken, setAccessToken] = useState(null);
  const [userProfile, setUserProfile] = useState(null);
  const [authError, setAuthError] = useState(null);
  const [isAuth0Used, setIsAth0Used] = useState(false);

  const [selectedLoginType, setSelectedLoginType] = useState(null);

  const changeSelectedLoginType = useCallback(
    (value) => {
      setSelectedLoginType(value);
    },
    [setSelectedLoginType],
  );

  const handleAuthError = useCallback((err) => {
    setAuthError(err);
    setAccessToken(null);
  }, []);

  /**
   * The main point for fetching a relevant access token.
   * This will return either access token if there is some or null
   * Also might throw an error if there is one during performing an API call to auth provider
   */
  const getAccessToken = useCallback(
    () =>
      /**
       * We have an option here. We can use either the `getToken` function from the context
       * ot use the static method `awaitGetToken` if we want to use the user management functionality
       * without context
       */
      //   NativeUserManagementContext.awaitGetToken()
      //     .then((getToken) => getToken())
      //     .then((token) => setAccessToken(token))
      //     .catch(handleAuthError),
      // [setAccessToken, handleAuthError],
      getToken()
        .then((token) => setAccessToken(token))
        .catch(handleAuthError),
    [getToken, setAccessToken, handleAuthError],
  );

  const getProfile = useCallback(() => getUserProfile().then(setUserProfile).catch(handleAuthError), [getUserProfile, handleAuthError]);

  const checkIsLoggedInViaAuth0 = useCallback(
    (accessToken) => NativeUserManagementContext.awaitIsOwnToken().then((isOwnToken) => isOwnToken(accessToken)),
    [],
  );

  /**
   * Initial fetching of accessToken after the application is launched
   * If an access token is received than the user is logged in.
   * If no token received or event an error thrown than the user is not logged in.
   */

  useEffect(() => {
    getAccessToken();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (accessToken) {
      getProfile();
    } else {
      setUserProfile({});
    }
  }, [accessToken, getProfile, setUserProfile]);

  /**
   * Check is the access token issued by Auth0 or not.
   * Such check is useful in scenarios where we need to use several identity providers in one application
   */
  useEffect(() => {
    checkIsLoggedInViaAuth0(accessToken).then(setIsAth0Used);
  }, [accessToken, checkIsLoggedInViaAuth0]);

  const value = useMemo(() => {
    const authMethodFactory = (authMethod) => () =>
      authMethod()
        .then(() => {
          setSelectedLoginType({});
          setUserProfile({});
          getAccessToken();
        })
        .catch(handleAuthError);

    const performLoginViaBrowser = authMethodFactory(login);

    const performForceLoginViaBrowser = authMethodFactory(forceLogin);

    const performLogoutViaBrowser = authMethodFactory(logout);

    const performLogin = authMethodFactory(loginViaWebView);

    const performForceLogin = authMethodFactory(forceLoginViaWebView);

    const performLogout = selectedLoginType === LOGIN_TYPES.embedded ? authMethodFactory(logoutViaEmbedded) : authMethodFactory(logoutViaWebView);

    const performStartLoginEmbedded = authMethodFactory(startLoginEmbedded);

    const performCompleteLoginEmbedded = authMethodFactory(completeLoginEmbedded);

    const passwordlessStart = (creds) => startLoginEmbedded(creds).then(console.log).catch(handleAuthError);

    const passwordlessLogin = (creds) => completeLoginEmbedded(creds).then(() => getAccessToken());

    const saveCredsEmbedded = (creds) =>
      saveCredentialsEmbedded(creds)
        .then(() => getAccessToken())
        .catch(handleAuthError);

    const updateProfile = (data) =>
      updateUserProfile(data)
        .then(() => getProfile())
        .catch(handleAuthError);

    return contextFactory({
      accessToken,
      login: performLogin,
      logout: performLogout,
      forceLogin: performForceLogin,
      loginViaBrowser: performLoginViaBrowser,
      logoutViaBrowser: performLogoutViaBrowser,
      forceLoginViaBrowser: performForceLoginViaBrowser,
      userProfile,
      setUserProfile,
      updateUserProfile: updateProfile,
      isAuth0Used,
      // saveCredentialsViaEmbedded: saveCredsEmbedded,
      // startLoginViaEmbedded: startLoginEmbedded,
      // completeLoginViaEmbedded: passwordlessLogin,
      getToken,
      setEnabledAnalytics,
      selectedLoginType,
      changeSelectedLoginType,
    });
  }, [
    login,
    forceLogin,
    logout,
    loginViaWebView,
    forceLoginViaWebView,
    selectedLoginType,
    logoutViaEmbedded,
    logoutViaWebView,
    startLoginEmbedded,
    completeLoginEmbedded,
    accessToken,
    userProfile,
    isAuth0Used,
    getToken,
    setEnabledAnalytics,
    changeSelectedLoginType,
    handleAuthError,
    getAccessToken,
    saveCredentialsEmbedded,
    updateUserProfile,
    getProfile,
  ]);

  return <context.Provider value={value}>{children}</context.Provider>;
};

AuthContext.context = context;

export { AuthContext };
