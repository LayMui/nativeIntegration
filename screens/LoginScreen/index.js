// import {AuthorizationUI} from '@yaradigitallabs/sh-user-management-native';
import CheckBox from "@react-native-community/checkbox";
import React, { useContext, useEffect, useState } from "react";
import { Button, StyleSheet, View, Text, ScrollView, Alert } from "react-native";
import { useNavigation } from "@react-navigation/core";
import { Platform } from "react-native";

import { AuthContext } from "../../contexts/auth";
import routes, { LOGIN_TYPES } from "../../constants/routes";

import testProperties from "../../utils/testProperties";
import { appData, regionData, envData, langData } from "../../utils/data";

import { RNSelectPicker } from "@yaradigitallabs/ahua-react-native";

import SelectPicker from "react-native-select-picker-dropdown";
const LoginScreen = ({
  navigation,
  notification,
  selectedEnv,
  onEnvSelected,
  language,
  onSetLanguage,
  selectedRegion,
  onRegionSelected,
  selectedApp,
  onAppSelected,
  rememberAuthSession,
  enableInnerSkipLogin,
  enableSkipLogin,
  onRememberAuthSessionUpdated,
  onEnableInnerSkipLogin,
  onEnableSkipLogin,
}) => {
  const { login, logout, forceLogin, logoutViaBrowser, forceLoginViaBrowser, changeSelectedLoginType } = useContext(AuthContext.context);

  const { navigate } = useNavigation();

  useEffect(() => {
    console.log("language: " + language);
    console.log("env: " + selectedEnv);
    console.log("region: " + selectedRegion);
    console.log("app: " + selectedApp);
  }, [language, selectedEnv, selectedRegion, selectedApp]);

  useEffect(() => {
    notification && Alert.alert(notification);
  }, [notification]);

  const [selected, setSelected] = useState();
  return (
    <ScrollView
      // eslint-disable-next-line react-native/no-inline-styles
      contentContainerStyle={{
        flexGrow: 1,
      }}
    >
      <View style={styles.container}>
        <Text style={{ ...styles.sectionHeader }}>Language code</Text>
        {/* <RNSelectPicker
          {...testProperties("langInput")}
          pickerOptions={langData}
          type={"flat"}
          defaultValue={langData.find((l) => l.value === language)}
          onValueChange={(val) => {
            if (Platform.OS === "android") {
              return onSetLanguage(val.newValue.value);
            }
            language = val.newValue.value;
          }}
          onBlur={() => {
            if (Platform.OS === "ios") {
              onSetLanguage(language);
            }
          }}
      /> */}
        <SelectPicker
               pickerProps={{
                  ...testProperties("langInput")
                }}
              
                placeholder={{label: 'Select a language'}}
                onUpArrow={(selected) => setSelected(selected - 1 )}
                onDownArrow={(selected) => setSelected(selected + 1)}
                onValueChange={(language) =>  onSetLanguage(language)}
                items={langData}
             />


        <Text style={styles.sectionHeader}>Available envs</Text>
        {/* <SelectPicker
          {...testProperties("envInput")}
          pickerOptions={envData}
          type={"flat"}
          defaultValue={envData.find((e) => e.value === selectedEnv)}
          onValueChange={(val) => {
            if (Platform.OS === "android") {
              return onEnvSelected(val.newValue.value);
            }

            selectedEnv = val.newValue.value;
          }}
          onBlur={() => {
            if (Platform.OS === "ios") {
              onEnvSelected(selectedEnv);
            }
          }}
        /> */}
           <SelectPicker
               pickerProps={{
                  ...testProperties("envInput")
                }}
                placeholder={{label: 'Select a environment'}}
                onValueChange={(selectedEnv) =>  onEnvSelected(selectedEnv)}
                items={envData}
             />

        <Text style={styles.subHeader}>Regions</Text>
        {/* <SelectPicker
          {...testProperties("regionInput")}
          pickerOptions={regionData}
          type={"flat"}
          defaultValue={regionData.find((r) => r.value === selectedRegion)}
          onValueChange={(val) => {
            if (Platform.OS === "android") {
              return onRegionSelected(val.newValue.value);
            }

            selectedRegion = val.newValue.value;
          }}
          onBlur={() => {
            if (Platform.OS === "ios") {
              onRegionSelected(selectedRegion);
            }
          }}
        /> */}
          <SelectPicker
               pickerProps={{
                  ...testProperties("regionInput")
                }}
                placeholder={{label: 'Select a region'}}
                onValueChange={(selectedRegion) =>  onRegionSelected(selectedRegion)}
                items={regionData}
             />

        <Text style={styles.subHeader}>Apps</Text>
        {/* <SelectPicker
          {...testProperties("appInput")}
          pickerOptions={appData}
          type={"flat"}
          defaultValue={appData.find((a) => a.value === selectedApp)}
          onValueChange={(val) => {
            if (Platform.OS === "android") {
              return onAppSelected(val.newValue.value);
            }

            selectedApp = val.newValue.value;
          }}
          onBlur={() => {
            if (Platform.OS === "ios") {
              onAppSelected(selectedApp);
            }
          }}
        /> */}
        <SelectPicker
            pickerProps={{
              ...testProperties("appInput")
            }}  
        
            placeholder={{label: 'Select an app'}}
            onValueChange={(selectedApp) =>  onAppSelected(selectedApp)}
            items={appData}
        />

    
        <View style={styles.row}>
       
          <Text>{"\n"}Remember auth session:</Text>
          <View>
            <CheckBox value={rememberAuthSession} onValueChange={onRememberAuthSessionUpdated} {...testProperties("authCheck")} />
          </View>
        </View>
        <View style={styles.row}>
        
          <View style={styles.row}>
            <Text>InnerSkipLogin:</Text>
            <View>
              <CheckBox value={enableInnerSkipLogin} onValueChange={onEnableInnerSkipLogin} {...testProperties("enableInnerSkipLogin")} />
            </View>
          </View>
          <View style={styles.row}>
            <Text>SkipLogin:</Text>
            <View>
              <CheckBox value={enableSkipLogin} onValueChange={onEnableSkipLogin} {...testProperties("enableSkipLogin")} />
            </View>
          </View>
        </View>
        <View style={{ height: 270, justifyContent: "flex-start" }}>
          {[
            (props) => (
              <Button
                {...props}
                key="Force log in via browser"
                title="Force log in via browser"
                {...testProperties("forceLoginViaBrowser")}
                onPress={forceLoginViaBrowser}
              />
            ),
            (props) => (
              <Button
                {...props}
                key="Logout via browser"
                title="Logout via browser"
                {...testProperties("logoutViaBrowser")}
                onPress={logoutViaBrowser}
              />
            ),
            (props) => <Button {...props} key="Log in" title="Log in" onPress={login} {...testProperties("login")} />,
            (props) => <Button {...props} key="Force log in" title="Force log in" {...testProperties("forceLogin")} onPress={forceLogin} />,
            (props) => <Button {...props} key="Log out" title="Log out" {...testProperties("logOut")} onPress={logout} />,
          ].map((Btn, index) => (
            <Btn key={index} color="#000" />
          ))}
        </View>
      </View>
    </ScrollView>
  );
};

export default LoginScreen;

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: "row",
  },
  container: {
    flex: 1,
    justifyContent: "space-around",
    paddingHorizontal: 20,
  },
  sectionHeader: {
    textAlign: "center",
    marginBottom: 10,
    textTransform: "uppercase",
    fontSize: 16,
    marginTop: 10,
  },
  subHeader: {
    textTransform: "uppercase",
    textAlign: "center",
    marginBottom: 10,
  },
  dropdown: {
    height: 50,
    borderColor: "gray",
    borderWidth: 0.5,
    borderRadius: 8,
    paddingHorizontal: 8,
  },
  icon: {
    marginRight: 5,
  },
  label: {
    position: "absolute",
    backgroundColor: "white",
    left: 22,
    top: 8,
    zIndex: 999,
    paddingHorizontal: 8,
    fontSize: 14,
  },
  placeholderStyle: {
    fontSize: 16,
  },
  selectedTextStyle: {
    fontSize: 16,
  },
  iconStyle: {
    width: 20,
    height: 20,
  },
  inputSearchStyle: {
    height: 40,
    fontSize: 16,
  },
});
