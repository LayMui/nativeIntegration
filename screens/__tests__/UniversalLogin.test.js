import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import LoginScreen from "../LoginScreen";
import { AuthContext } from "../../contexts/auth";

import { NativeUserManagementContext } from "@yaradigitallabs/sh-user-management-native";

describe("Login Page With Universal Auth0 Login", () => {
  let instance;
  const storage = AsyncStorage;
  const selectedEnv = "local";
  const DEFAULT_PROPS = {
    authProviderConfig: {
      ["local"]: {
        domain: "SOME_DOMAIN",
        clientId: "SOME_CLIENT_ID",
      },
    },
    LOGIN_PROPS: {
      selectedEnv: "local",
      language: "english",
      selectedRegion: "india",
      selectedApp: "farmCare",
      rememberAuthSession: false,
      notification: false,
    },
  };

  afterEach(cleanup);

  beforeEach(() => {
    instance = (
      <NavigationContainer>
        <NativeUserManagementContext authProviderConfig={DEFAULT_PROPS.authProviderConfig} tenant={selectedEnv} storage={AsyncStorage}>
          <AuthContext>
            <LoginScreen {...DEFAULT_PROPS.LOGIN_PROPS} />
          </AuthContext>
        </NativeUserManagementContext>
      </NavigationContainer>
    );
  });

  test("Matches snapshots", () => {
    const snapshot = render(instance).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
