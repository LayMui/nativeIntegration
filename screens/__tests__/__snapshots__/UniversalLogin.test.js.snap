// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page With Universal Auth0 Login Matches snapshots 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "space-around",
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Language code
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="langInput"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              editable={false}
              style={
                [
                  undefined,
                  {
                    "color": "#c7c7cd",
                  },
                ]
              }
              testID="text_input"
              value="Select a language"
            />
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Available envs
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="envInput"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              editable={false}
              style={
                [
                  undefined,
                  {
                    "color": "#c7c7cd",
                  },
                ]
              }
              testID="text_input"
              value="Select a environment"
            />
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "marginBottom": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Regions
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="regionInput"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              editable={false}
              style={
                [
                  undefined,
                  {
                    "color": "#c7c7cd",
                  },
                ]
              }
              testID="text_input"
              value="Select a region"
            />
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "marginBottom": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Apps
      </Text>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="appInput"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              editable={false}
              style={
                [
                  undefined,
                  {
                    "color": "#c7c7cd",
                  },
                ]
              }
              testID="text_input"
              value="Select an app"
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text>
          

          Remember auth session:
        </Text>
        <View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              {
                "checked": false,
                "disabled": false,
              }
            }
            accessibilityValue={
              {
                "text": "off",
              }
            }
            accessible={true}
            pointerEvents="auto"
            testID="authCheck"
          >
            <RNCCheckbox
              accessibilityLabel="authCheck"
              forwardedRef={null}
              onValueChange={[Function]}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  undefined,
                ]
              }
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text>
            InnerSkipLogin:
          </Text>
          <View>
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="enableInnerSkipLogin"
            >
              <RNCCheckbox
                accessibilityLabel="enableInnerSkipLogin"
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text>
            SkipLogin:
          </Text>
          <View>
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="enableSkipLogin"
            >
              <RNCCheckbox
                accessibilityLabel="enableSkipLogin"
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "height": 270,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessibilityLabel="forceLoginViaBrowser"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="forceLoginViaBrowser"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Force log in via browser
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="logoutViaBrowser"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="logoutViaBrowser"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Logout via browser
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="login"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="login"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Log in
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="forceLogin"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="forceLogin"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Force log in
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="logOut"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="logOut"
        >
          <View
            style={
              [
                {},
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Log out
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
